### PR TITLE
fix(pipeline-stop): Fetch latest resource version to avoid 409 conflict

### PR DESCRIPTION
**Pull request by abhinandan13jan**
Addresses https://jira.coreos.com/browse/ODC-1898
Screencast from 11-18-2019 04_51_41 PM

### CODE REVIEW

File: frontend/packages/dev-console/src/utils/pipeline-actions.tsx [hunk 1]
```diff
@@ -175,10 +175,17 @@ export const stopPipelineRun: KebabAction = (kind: K8sKind, pipelineRun: Pipelin
...
     label: 'Stop',
     callback: () => {
-      k8sUpdate(PipelineRunModel, {
-        ...pipelineRun,
-        spec: { ...pipelineRun.spec, status: 'PipelineRunCancelled' },
-      });
+      // Fetching the latest resourceVersion for the Pipelinerun to avoid 409 conflict
+      k8sGet(PipelineRunModel, pipelineRun.metadata.name, pipelineRun.metadata.namespace)
+        .then((latestPLR) =>
+          k8sUpdate(PipelineRunModel, {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/utils/pipeline-actions.tsx]: Why is this not using `patch`?

We don't want to get into a pattern of fetching resources before updating. It doesn't actually solve any problem. Only further mitigates an already low percent chance of error. Actually we already do this when firehose is being used.

Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/utils/pipeline-actions.tsx]: @christianvogt Updated to use patch. Works well after removing the resourceVersion from the equation.

File: frontend/packages/dev-console/src/utils/pipeline-actions.tsx [hunk 2]
```diff
@@ -175,10 +175,19 @@ export const stopPipelineRun: KebabAction = (kind: K8sKind, pipelineRun: Pipelin
...
+        PipelineRunModel,
+        {
+          metadata: { name: pipelineRun.metadata.name, namespace: pipelineRun.metadata.namespace },
+        },
+        [
+          {
+            op: 'replace',
+            path: `/spec`,
+            value: { ...pipelineRun.spec, status: 'PipelineRunCancelled' },
+          },
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/utils/pipeline-actions.tsx]: A patch request should be as small as possible only patching the parts of the resource you are changing.
```
          {
            op: 'replace',
            path: `/spec/status`,
            value: 'PipelineRunCancelled',
          },
```

Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/utils/pipeline-actions.tsx]: updated, the only problem that could occur though is, If someone starts a PipelineRun and stops it immediately, before the status field is generated, the patch might not work, but it is a very edge case.

**Pull request merged**